### PR TITLE
Scripts/VoA: Moved scripts to SAI.

### PR DESCRIPTION
**Pull request by HoustonCore**
**Changes proposed:**
- Move all script mobs Vault of Archavon to SAI.

**Target branch(es):** 3.3.5

- [√ ] 3.3.5

**Tests performed:**

Build, tested in-game.
- Need more testing and feedback

### CODE REVIEW

**Comment by Shauren** [comment]
Can everyone please stop moving scripts between c++ and sai all the time?

**Comment by HoustonCore** [comment]
> Can everyone please stop moving scripts between c++ and sai all the time?

What happened this time? As you said, we spent 10 years trying to get rid of unnecessary code, and here we are.

File: sql/updates/world/3.3.5/2020_xx_xx_xx_world.sql [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+(32353, 0, 1, 0, 0, 0, 100, 4, 5000, 7000, 5000, 12000, 0, 11, 60899, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Archavon Warder - In Combat - Cast \'Shield Crush\' (Heroic Dungeon)'),
+(32353, 0, 2, 0, 0, 0, 100, 2, 8000, 10000, 17000, 20000, 0, 11, 60902, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Archavon Warder - In Combat - Cast \'Whirl\' (Normal Dungeon)'),
+(32353, 0, 3, 0, 0, 0, 100, 4, 8000, 10000, 17000, 20000, 0, 11, 60916, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Archavon Warder - In Combat - Cast \'Whirl\' (Heroic Dungeon)'),
+(32353, 0, 4, 0, 0, 0, 100, 2, 10000, 15000, 15000, 25000, 0, 11, 60919, 0, 0, 0, 0, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, 'Archavon Warder - In Combat - Cast \'Rock Shower\' (Normal Dungeon)'),
+(32353, 0, 5, 0, 0, 0, 100, 4, 10000, 15000, 15000, 25000, 0, 11, 60923, 0, 0, 0, 0, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, 'Archavon Warder - In Combat - Cast \'Rock Shower\' (Heroic Dungeon)');
+
+-- Flame Warder
+UPDATE `creature_template` SET `AIName`= 'SmartAI',`ScriptName`="" WHERE `entry` = 35143;
+
+DELETE FROM `smart_scripts` WHERE (`entryorguid` = 35143) AND (`source_type` = 0) AND (`id` IN (0, 1));
```
Comment (jackpoz) [inline sql/updates/world/3.3.5/2020_xx_xx_xx_world.sql:18]: are you supposed to keep any smart_script for entry 35143 with a different source_type or different id ?

Comment (HoustonCore) [reply sql/updates/world/3.3.5/2020_xx_xx_xx_world.sql:18]: No, I can only overwrite the deletion by entryorguid without touching the source_type and id.

Comment (HoustonCore) [reply sql/updates/world/3.3.5/2020_xx_xx_xx_world.sql:18]: No, I can only overwrite the deletion by entryorguid without touching the source_type and id.

**Comment by offl** [comment]
1. You need to remove all spells and other stuff used in scripts you want to remove
2. You changed timers and event types, for example one spell is used on aggro and in SAI script you cast it on respawn, are these changes based on something?
3. Some parts of scripts simply cannot be converted to SAI, for example https://github.com/TrinityCore/TrinityCore/pull/25419/files#diff-1308d4afe571259b738a88ce0cc918f4L207-L263

We can't lose and broke something during convertation, after convertation scripts are supposed to work better. Timers should be changed only if they are incorrect, same with spells, event types and other stuff

**Comment by HoustonCore** [comment]
> 1. You need to remove all spells and other stuff used in scripts you want to remove
> 2. You changed timers and event types, for example one spell is used on aggro and in SAI script you cast it on respawn, are these changes based on something?
> 3. Some parts of scripts simply cannot be converted to SAI, for example https://github.com/TrinityCore/TrinityCore/pull/25419/files#diff-1308d4afe571259b738a88ce0cc918f4L207-L263
> 
> We can't lose and broke something during convertation, after convertation scripts are supposed to work better. Timers should be changed only if they are incorrect, same with spells, event types and other stuff



Oh, all possible timers and actions were summed up from wowhead and from the logic of the fight, but they are not perfect for blizlyka. Some spels and events are used by bosses, but I can carefully sum it up. And if you are talking about health, then watch this video and compare your old code, is it true? 2020 year you rewrite functions, add new parameters for the future, change the names of functions, it's already Shadowlands and the battle with the ship did not work, and does not work

https://m.youtube.com/watch?v=_SrBVN4mHdo#menu

**Comment by Aokromes** [comment]
battlegunship? you mean icecrown gunship? if no crossfaction it works almost fine.

**Comment by HoustonCore** [comment]
"Almost" fine, you all do your job well, but you have a mania for standards. You change the functions of the software several times a year, when it already works. How many good PRS have been opened and it would work perfectly, but you don't like how it's done. That's why most of them switched to Azerothcore.

**Comment by jackpoz** [comment]
Could you please focus on this particular pr and on the review that offl left ? You can continue the discussion about TC on the forum

**Comment by offl** [comment]
Mostly we merge PRs really fastly, expecially if they are not big and if the proposed changes looks valid to at least one member.
But I understand what you're talking about, sometimes people ask for multiple not really needed changes and because of them PRs can be abandonned. Definitely not the case here and let me explain why.
My comment was about basic standarts(obviously if a script is removed, everything related to it must be removed too(e.g. spells used in script) also we can't just remove something with no reason, we can't convert something and lose some important parts of code) and not about codestyle standtarts you are talking about and I don't want to make those scripts "perfect".

I clearly understand that quite possible these CPP scripts are not entirely correct but we can't simply remove this part of code https://github.com/TrinityCore/TrinityCore/pull/25419/files#diff-1308d4afe571259b738a88ce0cc918f4L207-L263 without convertation to SAI and actually it's not possible to covert that specific script. So my suggestion is to remove all spells and other stuff used in scripts you want to remove and restore the script I mentioned above.
Later I can take a look at SAI scripts and edit them if needed, seems like they're quite simple and can be converted to DB.

**Comment by HoustonCore** [comment]
@offl Done check please. One question on Tempest Minion, and if we add emotion after the death and make a link summon creature. It's supposed to work.

**Comment by HoustonCore** [comment]
@offl Can you help me. Where you take info time event and cast spells? Sniff for blizzlike?

**Comment by offl** [comment]
Our current sniffs:
Sniffs (surprisingly) is not a good source for timers until you sniff everything by yourself and FOR yourself. Almost all our sniffs are normal gameplay, in other words all creatures are killed as fast as possible, nobody waited 5 minute to catch all repeat timers. Some packs of trash are skipped, some events are skipped or are incomplete, etc.

From retail (with or without sniffer, everyone can use combat log, it's actually easier than reading sniffs):
To catch initial timers you need to be **almost instantly** inside creature's melee range after aggro(depends on AI type, if it's hunter, he will cast only melee spells if you will be in melee range. Casters will behave differently if you will be in their melee range too). Repeat 10-15 times or so (depends on how correct values do you want).
Same for repeat timers, simply allow creature to cast spells on you.
Almost nobody does things like that so mostly timers are guessed. But this is how you make perfect script based on retail data. But you still can't catch timers and target types for all spells because some spells can be casted only on non-tank(mostly only in dungeons). And here comes our sniffs from normal gameplay. But again, it's mostly for dungeons.

Wowhead:
Some info can be taken from wowhead. I mean not just comments but also spells. Do not use spells from current wowhead for classic creatures because tons of them were changed. Even for WotLK or TBC current wowhead's data may be incomplete.
In this case we have http://web.archive.org/
Simply search for page like www.wowhead.com/npc=18760
You will be taken to http://web.archive.org/web/2019*/www.wowhead.com/npc=18760
Change year to 2010 and click on one of available archieved page
You will be taken to http://web.archive.org/web/20100415000000*/www.wowhead.com/npc=18760 for example
Almost nothing shown, rigth?
Right-click on page and click on option *View page source*
You will be taken to ```text view-source:http://web.archive.org/web/20100625141124/www.wowhead.com/npc=18760``` if you are using google chrome. I have no idea about other browsers
Type ctrl+f and type something you are looking for. For example for abilities. Or you can always check for Listview. It contains spells, drops, quests and other stuff.
In case of abilities all abilities will be look like that:
```text
new Listview({template: 'spell', id: 'abilities', name: LANG.tab_abilities, tabs: tabsRelated, parent: 'lkljbjkb574', visibleCols: ['school'], hiddenCols: ['reagents', 'skill'], computeDataFunc: Listview.funcBox.initModeFilter, onAfterCreate: Listview.funcBox.addModeIndicator, data: [{"cat":-8,"id":33844,"level":20,"name":"@Entangling Roots","school":3,modes:{"mode":4}},{"cat":-8,"id":15798,"level":20,"name":"@Moonfire","school":6,modes:{"mode":4}},{"cat":-8,"id":9739,"level":20,"name":"@Wrath","school":3,modes:{"mode":4}}]});
```

Videos:
Videos are good source for timers, expecially the old ones because the player's damage is low so creature will be not killed instantly.
For example this guy casts Overcharge at 40%, I also checked wowhead's comments, so the value is correct.
```sql
-- Tempest Warder (https://youtu.be/8hpdbio-n-Y?t=454)
UPDATE `creature_template` SET `AIName` = "SmartAI" WHERE `entry` = 34015;
DELETE FROM `smart_scripts` WHERE `entryorguid` = 34015 AND `source_type` = 0;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`event_param5`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_param4`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(34015,0,0,0,0,0,100,0,5000,8000,4000,7000,0,11,64363,0,0,0,0,0,5,0,0,0,0,0,0,0,0,"Tempest Warder - In Combat - Cast 'Shock'"),
(34015,0,1,2,2,0,100,1,0,40,0,0,0,11,64379,0,0,0,0,0,1,0,0,0,0,0,0,0,0,"Tempest Warder - Between 0-40% Health - Cast 'Overcharge' (No Repeat)"),
(34015,0,2,0,61,0,100,0,0,0,0,0,0,1,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,"Tempest Warder - On Link - Say Line 0");
```
This guy was fully sniffed, values are pretty accurate because I got tons of casts (creature wasn't killed fastly, wasn't skipped, also the sniff contained both modes - heroic and normal).
Auras can be seen in movies (they're displayed while creature is out of combat).
```sql
-- Frost Warder (3.3.5 Sniffs Normal & Heroic)
UPDATE `creature_template_addon` SET `auras` = "72122" WHERE `entry` = 38482;
UPDATE `creature_template_addon` SET `auras` = "71993" WHERE `entry` = 38483;
UPDATE `creature_template` SET `AIName` = "SmartAI", `ScriptName` = "" WHERE `entry` = 38482;
DELETE FROM `smart_scripts` WHERE `entryorguid` = 38482 AND `source_type` = 0;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`event_param5`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_param4`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(38482,0,0,0,0,0,100,2,5000,5000,5000,9000,0,11,72123,0,0,0,0,0,5,0,0,0,0,0,0,0,0,"Frost Warder - In Combat - Cast 'Frost Blast' (Normal Dungeon)"),
(38482,0,1,0,0,0,100,4,5000,5000,5000,9000,0,11,72124,0,0,0,0,0,5,0,0,0,0,0,0,0,0,"Frost Warder - In Combat - Cast 'Frost Blast' (Heroic Dungeon)");
```

Also there are ACID scripts, some of them are written with official data. Honestly I still don't know where that official data was taken from but those scripts are really good. Not perfect but better than scripts from videos or with guessed timers. The sad part is all official data scripts are done for vanilla and TBC. One day I will import all missing scripts from ACID's official data, expecially for remaining TBC heroics.
All other ACID's scripts too can be used because mostly they were done by well experienced developers.
TBC https://github.com/cmangos/tbc-db/blob/master/ACID/acid_tbc.sql
WotLK https://github.com/cmangos/wotlk-db/blob/master/ACID/acid_wotlk.sql

Also you can use existing scripts to make new. Why not.

I use different sources, mostly always they are wowhead and videos. Even if I have sniffs I still use wowhead and videos too because having only wowhead or videos or sniffs doesn't allow me to script something properly all the time.

**Comment by HoustonCore** [comment]
@offl Thanks for your guide, well, I almost did the right thing, though I'm the only one who recorded the fight on a stopwatch)

**Pull request merged**